topology/src/main/java/io/camunda/zeebe/topology/gossip/ClusterTopologyGossiper.java

```diff
@@ -55,8 +57,14 @@ public ClusterTopologyGossiper(
     this.clusterTopologyUpdateHandler = clusterTopologyUpdateHandler;
   }
 
-  public void start() {
-    executor.run(this::internalStart);
+  public ActorFuture<Void> start() {
```

Comment:
Yes. Also to keep it decoupled with `ClusterTopologyManager`. Making it async lets us having two different actors for them. Otherwise, we have this implicit assumption that they are running on same thread.